packages/next/next-server/server/next-server.ts

```diff
@@ -786,6 +786,16 @@ export default class Server {
     query: ParsedUrlQuery = {},
     parsedUrl?: UrlWithParsedQuery
   ): Promise<void> {
+    if (!pathname.startsWith('/')) {
+      const noSlashError = `Cannot render page with path "${pathname}", did you mean "/${pathname}"?`
+
+      if (process.env.NODE_ENV !== 'production') {
```

Comment:
Let's maybe always warn to not break existing apps.

Reply:
Updated